fdbclient/MonitorLeader.actor.cpp

```diff
@@ -142,6 +154,38 @@ std::string ClusterConnectionString::getErrorString( std::string const& source,
 	}
 }
 
+ACTOR Future<Void> _resolveHostnames(ClusterConnectionString* self) {
+	std::vector<Future<Void>> fs;
+	for (auto const& hostName : self->mutableHostnames()) {
+		fs.push_back(map(INetworkConnections::net()->resolveTCPEndpoint(hostName.host, hostName.service),
+		                 [=](std::vector<NetworkAddress> const& addresses) -> Void {
+			                 NetworkAddress addr = addresses[deterministicRandom()->randomInt(0, addresses.size())];
```

Comment:
Why do we select a random one here?

Reply:
I don't remember, TBH. But most likely it was because the host name got resolved into multiple addresses and it just randomly pick one